internal/network/machine_network_cidr.go

```diff
@@ -417,6 +418,64 @@ func GetInventoryNetworksByFamily(hosts []*models.Host, log logrus.FieldLogger)
 	return ret, nil
 }
 
+func GetDefaultRouteNetworkByFamily(h *models.Host, networks map[AddressFamily][]string, log logrus.FieldLogger) (map[AddressFamily]string, error) {
+	var err error
+	ret := make(map[AddressFamily]string)
+
+	//start with dummy route
+	defaultRoutev4 := &models.Route{Metric: math.MaxInt32}
+	defaultRoutev6 := &models.Route{Metric: math.MaxInt32}
+
+	if h.Inventory != "" {
+		var inventory models.Inventory
+		err = json.Unmarshal([]byte(h.Inventory), &inventory)
+		if err != nil {
+			log.WithError(err).Warnf("Unmarshal inventory %s", h.Inventory)
+			return ret, err
+		}
+		//find the minimal default route
+		var isDefault, found bool
+		for _, route := range inventory.Routes {
+			if isDefault, err = IsDefaultRoute(route); err != nil {
+				log.WithError(err).Errorf("fail to find default route")
+				return ret, err
+			}
+			if isDefault {
+				if IsIPv4Addr(route.Gateway) && route.Metric < defaultRoutev4.Metric {
+					log.Info("==> found ipv4 route")
+					defaultRoutev4 = route
+					found = true
+				} else if IsIPv6Addr(route.Gateway) && route.Metric < defaultRoutev6.Metric {
+					log.Info("==> found ipv6 route")
+					defaultRoutev6 = route
+					found = true
+				}
+			}
+		}
+
+		if !found {
+			return ret, fmt.Errorf("could not find default route")
+		}
+
+		//if the default route gateway is located in one of the cidrs
+		//in the inventory networks then return this network (per family)
+		for _, cidr := range networks[IPv4] {
+			if ipInCidr(defaultRoutev4.Gateway, cidr) {
+				ret[IPv4] = cidr
+				break
+			}
+		}
+		for _, cidr := range networks[IPv6] {
+			if ipInCidr(defaultRoutev6.Gateway, cidr) {
+				ret[IPv6] = cidr
+				break
+			}
+		}
+		return ret, nil
```

Comment:
If default route was found for only 1 family, is it OK to return the map for both ?

Reply:
It is necessary because the calling function is working per family. in the calling function we check per family if we have an input and proceed accordingly